src/forms/inputcounter/InputCounter.tsx

```diff
@@ -0,0 +1,134 @@
+        this.state = {
+            value: defaultValue,
+            errorMsg: "",
+        };
+    }
+
+    private getErrorMessage() {
+        if (this.props.errMsg === undefined) {
+            return "Value should be ";
+        } else {
+            return this.props.errMsg;
+        }
+    }
+
+    private handleChange = (evt: React.ChangeEvent<HTMLInputElement>) => {
+        if (isNaN(parseInt(evt.target.value))) {
+            this.setState({
+                value: isNaN(this.props.minValue!) ? 0 : this.props.minValue!,
+            });
+        } else if (parseInt(evt.target.value) > this.props.maxValue!) {
+            let maxValueError = this.getErrorMessage() + " <= " + this.props.maxValue;
+            this.setState({errorMsg: maxValueError});
+            this.setState({value: parseInt(evt.target.value)});
+        } else if (parseInt(evt.target.value) < this.props.minValue!) {
+            let minValueError = this.getErrorMessage() + " >= " + this.props.minValue;
+            this.setState({errorMsg: minValueError});
+            this.setState({value: parseInt(evt.target.value)});
+        } else {
+            this.setState({value: parseInt(evt.target.value)});
+            this.setState({errorMsg: ""});
+        }
+        const {onChange} = this.props;
+        if (onChange) onChange(evt);
+    };
+
+    private handleIncrement() {
+        if (this.props.maxValue === undefined) {
+            this.setState({value: this.state.value + 1});
+        } else if (this.state.value < this.props.maxValue!) {
+            this.setState({value: this.state.value + 1});
+        } else {
+            let maxValueError = this.getErrorMessage() + " <= " + this.props.maxValue;
+            this.setState({errorMsg: maxValueError});
+        }
+    }
+
+    private handleDecrement() {
+        if (this.props.minValue === undefined) {
+            this.setState({value: this.state.value - 1});
+        } else if (this.state.value > this.props.minValue!) {
+            this.setState({value: this.state.value - 1});
+        } else {
+            let minValueError = this.getErrorMessage() + " >= " + this.props.minValue;
+            this.setState({errorMsg: minValueError});
+        }
+    }
+
+    render() {
+        const {maxValue, minValue, defaultValue, onChange, name, errMsg} = this.props;
+        let {width} = this.props;
+        if (width === undefined) {
+            width = "6rem"; //This is defaultWidth
+        }
+        return (
+            <div className="clr-form" style={{width: width}}>
+                <div style={{display: "flex"}}>
+                    <div style={{display: "inline-block", paddingRight: "0%", paddingLeft: "0%"}}>
+                        <input
+                            name="counter"
+                            value={this.state.value}
+                            onChange={this.handleChange.bind(this)}
+                            style={{height: "52px", width: "100%", border: "1px solid #d3d3d3"}}
```

Comment:
Same here, the styles can be added to a css file.

Reply:
Addressed.